grails-app/controllers/org/pih/warehouse/inventory/StockMovementController.groovy

```diff
@@ -267,7 +268,30 @@ class StockMovementController {
     }
 
     def rollbackApproval = {
-        throw new NotImplementedException("Action ${params.actionName} not implemented")
+        String stockMovementId = params.get("id")
+        if (!stockMovementService.canRollbackApproval(AuthService.currentUser.get(), stockMovementId)) {
+            flash.message = g.message(
+                    code: "request.errors.approval.rollback.message",
+                    default: "Unable to rollback approval: You do not have permissions to perform this action",
+                    args: [g.message(code: "errors.noPermissions.label")]
+            )
+        }
+
+        try {
```

Comment:
I wish we were full React, not to have those ugly try/catches to display an error message.